Source/JavaScriptCore/API/JSContextRefPrivate.h

```diff
@@ -94,6 +94,30 @@ JS_EXPORT void JSContextGroupSetExecutionTimeLimit(JSContextGroupRef group, doub
 */
 JS_EXPORT void JSContextGroupClearExecutionTimeLimit(JSContextGroupRef group) JSC_API_AVAILABLE(macos(10.6), ios(7.0));
 
+/*!
+@function
+@abstract Enables sampling profiler.
+@param group The JavaScript context group that the sampling profiler gets enabled.
+@result The value of the enablement, true if the sampling profiler gets enabled, otherwise false.
+@discussion Remote inspection is true by default.
+*/
+JS_EXPORT bool JSContextGroupEnableSamplingProfiler(JSContextGroupRef group) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+@function
+@abstract Disables sampling profiler.
+@param group The JavaScript context group that the sampling profiler gets disabled.
+*/
+JS_EXPORT void JSContextGroupDisableSamplingProfiler(JSContextGroupRef group) JSC_API_AVAILABLE(macos(JSC_MAC_TBA), ios(JSC_IOS_TBA));
+
+/*!
+@function
+@abstract Gets sampling profiler output in JSON form and clears the sampling profiler records.
+@param group The JavaScript context group that the sampling profiler output is taken.
```

Comment:
I would say "The JavaScript context group whose sampling profile output is taken"